src/test/java/com/squareup/kotlinpoet/FunSpecTest.kt

```diff
@@ -186,6 +186,23 @@ class FunSpecTest {
       |""".trimMargin())
   }
 
+  @Test fun functionParamWithKdocAndReturnKdoc() {
+    val funSpec = FunSpec.builder("foo")
+        .addParameter(ParameterSpec.builder("string", String::class.asTypeName())
+            .addKdoc("A string parameter.\n")
+            .build())
+        .addParameter(ParameterSpec.builder("nodoc", Boolean::class.asTypeName()).build())
+        .returns(String::class.asTypeName(), CodeBlock.of("the foo.\n"))
```

Comment:
Yeah, woulda been a lot nicer to write `returns(String::class, kdoc = "the foo")`